src/Symfony/Component/TypeInfo/TypeResolver/StringTypeResolver.php

```diff
@@ -223,7 +223,17 @@ private function getTypeFromNode(TypeNode $node, ?TypeContext $typeContext): Typ
         }
 
         if ($node instanceof UnionTypeNode) {
-            return Type::union(...array_map(fn (TypeNode $t): Type => $this->getTypeFromNode($t, $typeContext), $node->types));
+            $types = [];
+
+            foreach ($node->types as $nodeType) {
+                $types[] = $type = $this->getTypeFromNode($nodeType, $typeContext);
```

Comment:
I suggest moving `$types[] = $type;` after the `if`, to make the code easier to understand (not putting `mixed` at all in the list of types)